Kernel/Syscalls/statvfs.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2021, Justin Mietzner <sw1tchbl4d3@sw1tchbl4d3.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/FileSystem/Custody.h>
+#include <Kernel/FileSystem/FileDescription.h>
+#include <Kernel/FileSystem/VirtualFileSystem.h>
+#include <Kernel/Process.h>
+
+namespace Kernel {
+
+KResultOr<int> Process::do_statvfs(String path, statvfs* buf)
+{
+    RefPtr<Custody> parent_custody;
+    auto custody_or_error = VFS::the().resolve_path(path, current_directory(), &parent_custody, 0);
+    if (custody_or_error.is_error())
+        return custody_or_error.error();
+
+    statvfs kernelbuf = {};
+    size_t longest_match = 0;
+
+    VFS::the().for_each_mount([&path, &kernelbuf, &longest_match](auto& mount) {
+        String mountpath = mount.absolute_path();
+
+        if (path.contains(mountpath) && mountpath.length() > longest_match) {
+            longest_match = mountpath.length();
+
+            auto& fs = mount.guest_fs();
```

Comment:
Hmm, what's going on here? If you are trying to find the underlying filesystem for a given path, it would be better to do a basic path resolution (resolving the path to a custody via `VFS::resolve_path()` and then getting the filesystem from custody->inode().fs().

Reply:
That's what I tried to do, yes, probably oversaw that you can get the filesystem off of a custody. I'll change that in a bit.